### ISSUE TITLE
Invalid variable reference syntax for variable git:branch

### ISSUE THREAD

**Issue by hotgazpacho** [jacob-meacham/serverless-plugin-git-variables#13]
This plugin seems incompatible with the [`serverless-mocha-plugin`](https://github.com/nordcloud/serverless-mocha-plugin). Running `sls invoke test` with the git-variables plugin results in the following error:

`ServerlessError: Invalid variable reference syntax for variable git:branch. You can only reference env vars, options, & files. You can check our docs for more info.`

My custom section includes this:
```yaml
custom:
  git: ${git:branch} - ${git:describe} - ${git:sha1}
```

And my functions reference it like so:
```yaml
functions:
  graphql:
    name: ${self:provider.stage}-graphql
    description: GraphQL Handler (GET/POST /graphql) - ${self:custom.git}
    handler: functions/graphql.handler
```

Deploying works just fine. This only fails when trying to run my test suite.

**Comment by jacob-meacham** [comment]
@hotgazpacho I thought I'd commented earlier this week, but evidently I did not. I took a look at this, and it appears that the issue is an interaction with how the mocha plugin loads the environment for the tests and how this plugin populates the variables. Specifically, this plugin hijacks the getValueFromSource call on serverless.variables (see https://github.com/jacob-meacham/serverless-plugin-git-variables/blob/develop/src/index.js#L24). However, the mocha plugin creates a new set of variables and tries to populate with that - without the hijacked function (see https://github.com/nordcloud/serverless-mocha-plugin/blob/master/index.js#L202). One fix in this plugin would be to hijack the Variable constructor and appropriately set getValueFromSource. One could also fix this in the mocha plugin by changing how the new set of variables are constructed. Happy to take a PR with the above fix - thanks!
